sys/net/application_layer/dhcpv6/client.c

```diff
@@ -707,6 +718,23 @@ static void _solicit_servers(event_t *event)
     msg_len += _compose_elapsed_time_opt(time);
     msg_len += _compose_oro_opt((dhcpv6_opt_oro_t *)&send_buf[msg_len], oro_opts,
                                 ARRAY_SIZE(oro_opts));
+
+    #ifdef MUD_URL
+    char mud_url[] = MUD_URL;
+    if (strlen(mud_url) <= 253 && strlen(mud_url) > 0) {
```

Comment:
That should probably be a define. (`0xFF - sizeof(dhcpv6_opt_mud_url_t)`?)

Also, no need to check the length for being 0, rather add a config option to enable the MUD option.
The "RIOT way" to do this would probably be by introducing a pseudo-module `gnrc_dhcpv6_client_mud_url` (`makefiles/pseudomodules.inc.mk`), then you can check

```C
if (IS_USED(MODULE_GNRC_DHCPV6_CLIENT_MUD_URL)) {
    …
}
```

Also, since this string is set at compile-time and can't be changed, you might as well use an `assert()` here. This will crash the application, but since the string is not expected to change, we don't have to perform the check each time in release mode.

Reply:
Thank you for all your helpful comments! I have already started to implement them but had some minor problems I wanted to you ask you about. 

As you suggested I tried to added the MUD URL feature in `makefiles/pseudomodules.inc.mk` as a submodule like so:

```c
...
PSEUDOMODULES += gnrc_dhcpv6_%
PSEUDOMODULES += gnrc_dhcpv6_client_mud_url  # the option I added
PSEUDOMODULES += gnrc_ipv6_default
...
```

However, when trying to use the submodule as you suggested, i. e.

```
if (IS_USED(MODULE_GNRC_DHCPV6_CLIENT_MUD_URL)) {
...
}
```

the if condition seems to always evaluate to `true`. Is something else needed to be able to properly use this submodule? (Sorry if I am missing something obvious here.)

Reply:
I tried to implement your feedback on the length check in 3814b51.

Reply:
> However, when trying to use the pseudo-module as you suggested, i. e.
> 
> ```
> if (IS_USED(MODULE_GNRC_DHCPV6_CLIENT_MUD_URL)) {
> ...
> }
> ```
> 
> the if condition seems to always evaluate to `true`.

even when you are not using the `gnrc_dhcpv6_client_mud_url` module?

Reply:
I think now it works as intended :) I must have gotten something wrong when I wrote my comment above. Now the tests performed when using `make test` only succeed if the pseudomodule `gnrc_dhcpv6_client_mud_url` is included in the makefile.